packages/cadl-rlc-emitter/test/unit/generate/responses.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { assert } from "chai";
+import { emitResponsesFromCadl } from "../emitUtil.js";
+import { assertEqualContent } from "../testUtil.js";
+
+describe("Responses.ts", () => {
+  it("should generate property name with custome name", async () => {
+    const responses = await emitResponsesFromCadl(`
+        @doc("Metadata for long running operation status monitor locations")
+        model LongRunningStatusLocation {
+            @doc("The location for monitoring the operation state.")
+            @header("Operation-Location")
+            operationLocation: ResourceLocation<string>;
+        }
+        op read(): LongRunningStatusLocation;
+    `);
+    assert.ok(responses);
+    assertEqualContent(
+      responses!.content,
+      `
+    import { RawHttpHeaders } from "@azure/core-rest-pipeline";
+    import { HttpResponse } from "@azure-rest/core-client";
+    
+    export interface Read204Headers {
+      /** The location for monitoring the operation state. */
+      "operation-location": string;
+    }
+    
+    /** Metadata for long running operation status monitor locations */
+    export interface Read204Response extends HttpResponse {
+      status: "204";
+      headers: RawHttpHeaders & Read204Headers;
+    }`
+    );
+  });
+
+  it("should generate property name without custome name", async () => {
+    const responses = await emitResponsesFromCadl(`
+        @doc("Metadata for long running operation status monitor locations")
+        model LongRunningStatusLocation {
+            @doc("The location for monitoring the operation state.")
+            @header
+            operationLocation: ResourceLocation<string>;
```

Comment:
why are we generate "operation-location" for this case ?

Reply:
In this case we could both get the property name `operationLocation` and `operation-location` I just follow the openapi way to get the name here. It would be the former one.

see code below:
https://github.com/microsoft/cadl/blob/bb5847a37d4a7064d2445cc488c47448f99da6f4/packages/openapi3/test/return-types.test.ts#L6-L20